### DISCUSSION TITLE
Setting same-site-cookie property from environment/.env file

### DISCUSSION THREAD

**Discussion by luca-bassoricci** [quarkusio/quarkus#23963] [Q&A] [answered]
I have trouble setting `quarkus.http.same-site-cookie.Mercurio_cookie.value` value to `None` from environment property or .env file.
Using `application.properties` file works fine but from environment value is not setted (property is translated to `QUARKUS_HTTP_SAME_SITE_COOKIE_MERCURIO_COOKIE_VALUE`).
Any advice?

Best regards
Luca

**Comment by sberyozkin** [comment]
CC @radcortez

**Comment by radcortez** [comment, marked as answer]
The env property format is unable to fully represent `quarkus.http.same-site-cookie.Mercurio_cookie.value`, because we only have available a subset of the possible characters. So, for instance, if you pick `QUARKUS_HTTP_SAME_SITE_COOKIE_MERCURIO_COOKIE_VALUE`, it can represent, `quarkus.http.same.site.cookie.mercurio.cookie.value` or other combinations. In fact, you don't how if the env variable `_` should map to a `.` a `-` or a `_`.

In that situation, you need to include the original property name in the dotted format in another config source (like `application.properties`). This will allow Quarkus based on an already existent property that `QUARKUS_HTTP_SAME_SITE_COOKIE_MERCURIO_COOKIE_VALUE` is actually equivalent to `quarkus.http.same-site-cookie.Mercurio_cookie.value`.

If you are just using the `.env` file, the `.env` file does support the dotted properties format as well, you you can just add the original format there.

**Reply by gsmet** [reply]
I would have thought something like the following would be easier:
```
quarkus.http.same-site-cookie.Mercurio_cookie.value=${your-property:defaultValue}
```
And then, in the `.env`:
```
YOUR_PROPERTY=...
```

**Reply by radcortez** [reply]
That also works :)

But if you really want to use the original config name that is the way.

**Reply by luca-bassoricci** [reply]
I can't change deployed code.
The same app is deployed on Openshift two times: the first time with `same-site` hardcode to `strict` and the second deployment is configured via 'Environment' using `QUARKUS_HTTP_SAME_SITE_COOKIE_MERCURIO_COOKIE_VALUE` (not working) approach.
The `.env` file can't be used because create it using Openshift's ConfigMaps will clean the working directory (just a question: dotted format will work for `env` file if running OS is Linux?).
Currently I am deploying a copy of the original `application.properties` to `$PWD\config\` with property in dotted format but deploy a full copy of file is not what I want to do; are config sources cumulative? In this case deploying in `$PWD\config\` only the changing properties will produce the desidered result?

I hope I was clear.
Thanks for help

**Reply by radcortez** [reply]
> The `.env` file can't be used because create it using Openshift's ConfigMaps will clean the working directory (just a question: dotted format will work for `env` file if running OS is Linux?). 

Yes. The file is read as is, is just that it will try to match the environment variable format to the dotted format. If there are any dotted format keys, it will read them as if it was an `application.properties` file. You can mix both formats in the `.env` file.

> Currently I am deploying a copy of the original `application.properties` to `$PWD\config\` with property in dotted format but deploy a full copy of file is not what I want to do; are config sources cumulative? In this case deploying in `$PWD\config\` only the changing properties will produce the desidered result?

Yes, you can just deploy what you want to override. In this case, `application.properties` keys in the `config` folder will override the ones in `src/main/resources` if they exist. The `.env` file key override both. Check here for a more in-depth explanation: https://quarkus.io/guides/config-reference#configuration-sources
